api/query/cache/service/main.go

```diff
@@ -235,30 +232,19 @@ func init() {
 }
 
 func main() {
-	autoProjectID, err := metadata.ProjectID()
-	if err != nil {
-		log.Warningf("Failed to get project ID from metadata service")
-	} else {
-		if *projectID == "" {
-			log.Infof(`Using project ID from metadata service: "%s"`, *projectID)
-			*projectID = autoProjectID
-		} else if *projectID != autoProjectID {
-			log.Warningf(`Using project ID from flag: "%s" even though metadata service reports project ID of "%s"`, *projectID, autoProjectID)
-		} else {
-			log.Infof(`Using project ID: "%s"`, *projectID)
-		}
-	}
+	flag.Parse()
+	shared.InitProjectID()
 
 	log.Infof("Serving index with %d shards", *numShards)
 	// TODO: Use different field configurations for index, backfiller, monitor?
 	logger := log.StandardLogger()
 
-	idx, err = index.NewShardedWPTIndex(index.HTTPReportLoader{}, *numShards)
+	idx, err := index.NewShardedWPTIndex(index.HTTPReportLoader{}, *numShards)
```

Comment:
This is the problem. You are not assigning to the global `idx`, but creating a new local `idx` instead, which caused the readiness check to fail.

Reply:
Thanks again for spotting this. Digging through logs would never have found it.